Resolve Issue 95: Added SQL to prune 311 data to most recent 30 days

BenjaminMuscato opened the pull request:
Updated the SQL query for gathering 311 data, to limit the data returned
to be from the most recent 30 days.

Also removed the arbitrary limitation that had been on the 311 data,
which limited it to gather the 250 most recent records.

BenjaminMuscato committed:
Added SQL to prune 311 data to most recent 30 days

Updated the SQL query for gathering 311 data, to limit the data returned
to be from the most recent 30 days.

Also removed the arbitrary limitation that had been on the 311 data,
which limited it to gather the 250 most recent records.

vonbearshark commented on main.js:
```diff
@@ -286,7 +286,7 @@
         "311": {
             id: "40776043-ad00-40f5-9dc8-1fde865ff571",
-            primaryFiltering: "WHERE \"NEIGHBORHOOD\" LIKE '%Oakland' ORDER BY \"CREATED_ON\" DESC",
+            primaryFiltering: "WHERE \"NEIGHBORHOOD\" LIKE '%Oakland' AND \"CREATED_ON\" >= (NOW() - '30 day'::INTERVAL) ORDER BY \"CREATED_ON\" DESC",
```
So this works on all data sets from the WPRDC? That would be awesome

BenjaminMuscato replied:
Haven't tried it on the other data-sets, but I'll try it out and get back to you!

vonbearshark commented on main.js:
```diff
@@ -455,7 +455,7 @@
         Promise.all([
             fetchWPRDCData("Police", { limit: 250 }),
-            fetchWPRDCData("311", { limit: 250 }),
+            fetchWPRDCData("311"),
```
Could we rm the arbitrary limitation on the others, too? Or do we need to have the work-around there?

BenjaminMuscato replied:
If the updated ".primaryFilter" that I used for the 311 data-set works for the others, we shouldn't need this limit for any of them. I'll test it out and get back to you!

BenjaminMuscato committed:
Merge remote-tracking branch 'refs/remotes/Pitt-CSC/master'

vonbearshark commented:
@BenjaminMuscato did you get a chance to run your tests? Before you commit again, you'll need to merge in the recent changes, which separate some constants into another JS file.
Also, could you remove this block of code https://github.com/Pitt-CSC/PantherView/blob/master/main.js#L264 ? Since you've made it unnecessary. You can also remove the TODO comment on line 204, too.

BenjaminMuscato committed:
Revert "Added SQL to prune 311 data to most recent 30 days"

This reverts commit e9824ec8d36dfa065a207c9ffb3b8d3828a5757b.

BenjaminMuscato committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Issue-95---WPRDC-SQL-Date-Range

BenjaminMuscato committed:
Re-added the 311 Pruning, after resolving conflicts with Master.

BenjaminMuscato committed:
Added pruning into the other filtering fields

vonbearshark commented:
Looks great! If you take out those TODO comments I can merge it in!

BenjaminMuscato committed:
Removed "TODO" Statements

Removed the "TODO" statements about adding the SQL filtering. I left the
JS filtering in place for now, as it's also needed to filter by Day and
Week, even after the data is gathered from the database.

BenjaminMuscato commented:
@vonbearshark I've removed the "TODO" comments, I think the PR is ready to go.

I didn't touch any of the JS filtering that's currently in place, since I think we still need it in its current form to filter through the data even after we've gotten it from the database (to show by Day, Week, and Month in the GUI).

BenjaminMuscato committed:
Some cleanup

vonbearshark reviewed (approved):
This looks fantastic!

vonbearshark commented:
Great work, @BenjaminMuscato. Thanks a lot! This is a good addition
